Use puresript-servant in frontend and e2e tests

IvantheTricourne opened the pull request:
Closes #94.

IvantheTricourne committed:
Move marketclient interface to frontend

IvantheTricourne committed:
Namespace marketclient modules under Test

IvantheTricourne committed:
Use the same contracts definition in frontend

IvantheTricourne committed:
Remove Test namespace + import changes

IvantheTricourne committed:
Fix scripts

IvantheTricourne committed:
Fix e2e tests

IvantheTricourne committed:
Remove extra comments

IvantheTricourne commented:
**NOTE:** running any of the scripts now fail without running the server to get contract configs. Previously, scripts would just hang without a running server instance.

blinky3713 commented:
looks good so far, we will also want to use it here to replace this string way of templating requests
https://github.com/f-o-a-m/sample-nft-project/blob/master/frontend/src/App/API.purs#L47

IvantheTricourne committed:
wip: change signal getters

IvantheTricourne committed:
wip: converting between etypes to signal type

IvantheTricourne committed:
wip: some conversions

IvantheTricourne committed:
Impl getSignalsWithSale

IvantheTricourne committed:
Remove unneeded import

IvantheTricourne committed:
Make clientEnv once

Pass clientEnv to all components that need it

IvantheTricourne committed:
use top level cenv

IvantheTricourne committed:
use ps-servant clientM router

IvantheTricourne committed:
Use getM for getContracts

+ updates all call sites

IvantheTricourne committed:
Add types for signal market

IvantheTricourne committed:
Impl getSignalTokenWithHistory

IvantheTricourne commented:
All this needs is to add some `e2e` tests for the new getters. Changed issue #94 to reflect this.

Changing this PR to WIP until I complete those.

IvantheTricourne committed:
Remove clientEnv

IvantheTricourne committed:
Move getClientEnv to e2e

IvantheTricourne committed:
fixes

IvantheTricourne committed:
Fix decodeJsonSignalMarketHistoryEntry

IvantheTricourne committed:
Add/remove "http:"

IvantheTricourne committed:
Hardcode http

IvantheTricourne committed:
Remove hardcoded http

The value of `window.location.protocol` used to assemble a `ClientEnv`
is always suffixed with a `:`. Servant and ws uses are updated
accordingly.

IvantheTricourne committed:
Fix owner assigning issue

owner was accidentally set to the holder of the token (i.e., the Signal Market
address) when put up for sale. This sets the owner to the address of the address
who initiated the sale

IvantheTricourne committed:
naming

IvantheTricourne committed:
test new getters

IvantheTricourne committed:
geohash doesn't work
